packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,325 @@
+3. a mix of the above with formatting determined by the application ("1.000", "-0.030", "1024.000")
+4. numeric values with a static unit string appended before or after (e.g. "$4.23" or "15%")
+5. numeric values with custom unit logic. Examples:
+    - a file size column that could show the value 1000 as "1000 bytes" but the value 1024 as "1KB"
+    - an elapsed time column that could show 63 seconds as "00:01:03" or "1 minute, 3 seconds"
+6. date/time values formatted in various ways ("October 27", "yesterday", "2023-12-28 08:27")
+
+In all of the above cases:
+
+-   data should be sortable and groupable by its actual value, not the string representation
+-   text styling like font and alignment should be provided by Nimble
+-   columns should support i18n behaviors like decimal separators, date/time formats, and localized content
+-   there should be an option to show "placeholder" text if no value is specified
+-   there should be a way to show the full value of truncated text (likely via a title / tooltip on hover)
+
+We may not choose to support all of the above initially but we should design our solutions with these use cases in mind.
+
+### Background
+
+[Number column work item](https://github.com/ni/nimble/issues/1011)
+
+[Date/time column work item](https://github.com/ni/nimble/issues/1014)
+
+[Table Column API](../table-columns-hld.md)
+
+[Table Spec](../README.md)
+
+[Table Column Text Spec](./table-column-text-field.md)
+
+[Visual Design Spec](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/specs/)
+
+### Non-goals
+
+-   Combinations of the use cases listed above in a single column. This will be needed in cases where the source data isn't uniformly typed (e.g. SLE tag values or notebook outputs). This HLD focuses on uniform data types; future HLDs may discuss ways to configure multiple types of columns to be conditionally displayed together.
+-   Editable numbers. This is not supported by the text column yet either.
+-   Numeric formatting for `nimble-number-field`. While we may choose to expose similar APIs for its numeric formatting, the complexities of it being an editable input control make this out of scope for now.
+-   Customizing the styling of the column content (other than text alignment). This is not supported by the text column yet either.
+-   Enum and boolean values formatted as localized strings (0 -> "Fail", 1 -> "Pass"). These will likely use the [mapping column being discussed elsewhere](https://github.com/ni/nimble/pull/1220).
+
+---
+
+## Design
+
+### Summary
+
+Nimble will provide base classes that can be derived from to define columns that call a formatting function to render their data as text. Clients which require app-specific formatting logic to support above use cases like 5 (custom unit logic) will define custom columns in their application that derive from these base classes.
```

Comment:
The cases I'm currently envisioning SLE needing this are:
1. elapsed time ("3 m, 27 s")
    - SLE's format is wonky enough that I wouldn't want it in Nimble (though we could push to change it to a more standard format)
    - includes lots of localizable text ("minutes", "days", "seconds", etc) and other possible internationalization concerns (12 vs 24 hours? decimal separators?) though that may get better with #1090
2. file size ("1023 bytes" or "1 KB")
    - it's a bit special-purpose (I could be convinced otherwise)
    - it also has several localizable strings

But yeah, I agree we should encourage teams to discuss with us first. I will re-order this section.

Reply:
1. I think it'll be a common use case and we should push to a common format. We can piece it together along with localization of the units via the Intl api, i.e.: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat which could eliminate the localization burden altogether
2. It's probably not uncommon to want to display file size units in engineering applications / applications from NI. I wouldn't be against that column type in nimble.